python/lsst/pipe/tasks/finalizeCharacterization.py

```diff
@@ -767,3 +798,297 @@ def compute_psf_and_ap_corr_map(self, visit, detector, exposure, src, isolated_s
+
+
+class FinalizeCharacterizationDetectorTask(FinalizeCharacterizationTaskBase):
+    """Run final characterization per detector."""
+    ConfigClass = FinalizeCharacterizationDetectorConfig
+    _DefaultName = 'finalize_characterization_detector'
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        input_handle_dict = butlerQC.get(inputRefs)
+
+        band = butlerQC.quantum.dataId['band']
+        visit = butlerQC.quantum.dataId['visit']
+        detector = butlerQC.quantum.dataId['detector']
+
+        isolated_star_cat_dict_temp = {handle.dataId['tract']: handle
+                                       for handle in input_handle_dict['isolated_star_cats']}
+        isolated_star_source_dict_temp = {handle.dataId['tract']: handle
+                                          for handle in input_handle_dict['isolated_star_sources']}
+
+        isolated_star_cat_dict = {tract: isolated_star_cat_dict_temp[tract] for
+                                  tract in sorted(isolated_star_cat_dict_temp.keys())}
+        isolated_star_source_dict = {tract: isolated_star_source_dict_temp[tract] for
+                                     tract in sorted(isolated_star_source_dict_temp.keys())}
+
+        struct = self.run(
+            visit,
+            band,
+            detector,
+            isolated_star_cat_dict,
+            isolated_star_source_dict,
+            input_handle_dict['src'],
+            input_handle_dict['calexp'],
+        )
+
+        butlerQC.put(
+            struct.psf_ap_corr_cat,
+            outputRefs.finalized_psf_ap_corr_detector_cat,
+        )
+        butlerQC.put(
+            struct.output_table,
+            outputRefs.finalized_src_detector_table,
+        )
+
+    def run(self, visit, band, detector, isolated_star_cat_dict, isolated_star_source_dict, src, exposure):
+        """
+        Run the FinalizeCharacterizationDetectorTask.
+
+        Parameters
+        ----------
+        visit : `int`
+            Visit number.  Used in the output catalogs.
+        band : `str`
+            Band name.  Used to select reserved stars.
+        detector : `int`
+            Detector number.
+        isolated_star_cat_dict : `dict`
+            Per-tract dict of isolated star catalog handles.
+        isolated_star_source_dict : `dict`
+            Per-tract dict of isolated star source catalog handles.
+        src : `lsst.afw.table.SourceCatalog`
+            Src catalog.
+        exposure : `lsst.afw.image.Exposure`
+            Calexp exposure.
+
+        Returns
+        -------
+        struct : `lsst.pipe.base.struct`
+            Struct with outputs for persistence.
+
+        Raises
+        ------
+        NoWorkFound
+            Raised if the selector returns no good sources.
+        """
+        # We do not need the isolated star table in this task.
+        # However, it is used in tests to confirm consistency of indexes.
```

Comment:
It seems like it should be removed as a connection then, although if it's not worth the effort then fair enough.

Reply:
I don't want to upset the apple cart too much here...